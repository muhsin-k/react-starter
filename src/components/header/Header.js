import React from 'react';
import PropTypes from 'prop-types';

function Header(props) {
  return (
    <div>
      <nav>
        <div className="nav-wrapper">
          <a href="#/home" className="brand-logo center">
            {props.title}
          </a>
          <ul id="nav-mobile" className="left hide-on-med-and-down">
            <a href="#/home" className="waves-effect waves-light btn">
              TMDB
            </a>
          </ul>
        </div>
      </nav>
    </div>
  );
}
Header.propTypes = {
  title: PropTypes.string,
};
Header.defaultProps = {
  title: 'Trending Movies',
};

export default Header;
