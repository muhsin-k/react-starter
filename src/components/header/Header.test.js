import React from 'react';
import ReactDOM from 'react-dom';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Header from './Header';

enzyme.configure({
  adapter: new Adapter(),
});

describe('Header', () => {
  // Normal
  // This test mounts a component and checks that it doesn't throw an exception during rendering.
  test('renders without crashing[Normal Way]', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });
  // The difference between shallow() and mount() is that shallow() tests components in isolation
  // from the child components they render
  // while mount() goes deeper and tests a component's children
  // Enzyme
  it('renders without crashing[Enzyme Way Mount]', () => {
    enzyme.mount(<Header />);
  });
  it('renders without crashing[Enzyme Way Shallow]', () => {
    enzyme.shallow(<Header />);
  });
  it('renders title ', () => {
    // const wrapper = enzyme.shallow(<Header />);
    // const welcome = <a href="#/home" class="brand-logo center">Trending Movies in India</a>;
    // expect(wrapper.contains(welcome)).to.equal(true);
    // expect(wrapper.contains(welcome)).toEqual(true);
  });
  // Snapshot
  it('should render correctly[Snapshot Way]', () => {
    const output = enzyme.shallow(<Header title="Trending Movies in India" />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
