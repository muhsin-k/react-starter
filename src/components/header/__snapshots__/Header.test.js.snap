// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly[Snapshot Way] 1`] = `
<div>
  <nav>
    <div
      className="nav-wrapper"
    >
      <a
        className="brand-logo center"
        href="#/home"
      >
        Trending Movies in India
      </a>
      <ul
        className="left hide-on-med-and-down"
        id="nav-mobile"
      >
        <a
          className="waves-effect waves-light btn"
          href="#/home"
        >
          TMDB
        </a>
      </ul>
    </div>
  </nav>
</div>
`;
